core/src/Streamly/Internal/Data/Unbox.hs

```diff
@@ -362,48 +362,71 @@ class Unbox a where
     -- | Write an element of type "a" to a MutableByteArray given the byte
     -- index.
     --
-    -- IMPORTANT: The implementation of this interface may not check the bounds
-    -- of the array, the caller must not assume that.
+    -- IMPORTANT: The implementation of this interface does not check the
```

Comment:
Depends on how we or someone else implements it in their code.

Reply:
reverted the change.